### DISCUSSION TITLE
blog/2023-02-13-causal-inference-with-logistic-regression/

### DISCUSSION THREAD

**Discussion by giscus** [ASKurz/blogdown5#8] [Comments]
# blog/2023-02-13-causal-inference-with-logistic-regression/

Part 3 of the GLM and causal inference series.

https://timely-flan-2986f4.netlify.app/blog/2023-02-13-causal-inference-with-logistic-regression/

**Comment by sjwild** [comment]
I like this third one. I have very few substantive contributions here. A few minor comments:
* There are some typos. I did not write them down and could not find some of them on my second reading. Sigh.
* I suggest using a histogram (or two) when you compute $\tau_{ATE}$. I am referring specifically to the paragraph ending with this line: "...them summarize the results from the 1,000 iterations by their mean and standard deviation." I generally find things like histograms more informative than written summaries, but YMMV.
  * found a typo. Should be "then summarize"
* I like the discussion at the end around the ATE vs CATEs. 
* It might be worth noting (footnote maybe) that for logistic regression the terminology around ATEs can be confusing. Depending on what you read, they are also sometimes called the __average contrast__ or __averaged marginal effect__.

**Reply by ASKurz** [reply]
Your overall comments are very good to hear, @sjwild. This post is, by far, the one I've been the most worried about. It's the main reason I've asked for peer review. 

As to your fourth bullet, do you have any examples from the literature on people using the terms *average contrast* or *averaged marginal effect*?

**Reply by sjwild** [reply]
Alright. A few references for other terms that are used.
* [Mood](https://academic.oup.com/esr/article-abstract/26/1/67/540767) refers to both _Average Marginal Effects_ and $\Delta\{P}$. In the discussion on Average Marginal Effects, it may not be clear to readers that Mood is referring to continuous variables, not discrete ones. There is a section on discrete variables later, where $\Delta\{P}$ is used.
* [Gelman, Hill, and Vehtari](https://users.aalto.fi/~ave/ROS.pdf) use _difference in probabilities_ (pg. 225).
* [Stata](https://www.stata.com/manuals/rmargins.pdf) uses the term _discrete differences_ for discrete variables and _Average Marginal Effect_ for continuous ones. But they also note: "__partial effect__ and __average partial effect__. Some authors restrict the term _marginal effect_ to mean derivatives and use the term partial effect to denote discrete differences; see _marginal effect and average marginal effect_." 
* [Agresti and Tarantola](https://users.stat.ufl.edu/~aa/articles/agresti_tarantola.pdf) use _discrete change_ for discrete variables and _Average Marginal Effect_ for continuous ones. 
* [McCabe et al.](https://www.tandfonline.com/doi/abs/10.1080/00273171.2020.1868966) use _discrete differences_. They also say that "we can apply discrete differences to define a marginal effect as the difference between two points on a regression function" to further confuse terminology :)

So just a giant confusing jumble of jargon all around.

**Comment by ASKurz** [comment]
Okay, I believe I've addressed all of @sjwild's concerns (though there's probably a lingering typo or two). I'd still love to get another set of eyes on this post before publishing, though.

**Comment by isabellaghement** [comment]
Hey Solomon, 

Here is some feedback on this post for your consideration, in no particular order. 

1) Can you define the logit() function explicitly as logit(p) = p/(1-p) and clarify that p/(1 - p) = odds.  

2) Can you clarify that the ANOVA-type model aims to model the logit-transformed probability p that a subject's medical record indicated any STI testing at 6 weeks as a function solely of their treatment group (i.e., intervention group or control group).  In this model, $\beta_1$ is the linear effect of treatment group on the logit-transformed probability p described above; this effect is unadjusted for the effects of covariates.  Because the logit() function is a non-linear function of p, that means we are assuming a non-linear effect of treatment group on the p itself.  

3) Can you clarify that the ANCOVA-type model aims to model the logit-transformed probability p that a subject's medical record indicated any STI testing at 6 weeks as a function not only of their treatment group (i.e., intervention group or control group) but also other relevant covariates.  In this model, $\beta_1$ is the linear effect of treatment group on the logit-transformed probability p described above,  adjusted for the effects of covariates.  Because the logit() function is a non-linear function of p, that means we are assuming a non-linear effect of treatment group on the p itself which depends on the specific values of each of the covariates we adjusted for.  "Adjusted" on the logit (or log-odds) scale means "for a subject with particular values of the covariates included in the model"; "unadjusted" on the logit (or log-odds) scale means "ignoring the values of the covariates".   

The reason I suggested items 2) and 3) above is because I found the reader needs some intuition for each model and an understanding on what the models estimate and how that differs from one model to another.

The other thing that would be useful to add here is that binary logistic regression models such as these ANOVA-type and ANCOVA-type model have 3 scales on which the results could be presented (whether or not these models are fitted in a causal framework):  

1. logit scale 

2. odds scale 

3. probability scale 

The logit scale is symmetric about 0 and goes from -infinity to +infinity.  This is the scale on which such models are initially estimated but it is not as interpretable as the other two scales. 

The odds scale goes from 0 to +infinity and has the value 1 as its anchor point.  This scale is usually preferred by those who work in the medical field.  

The probability scale is bounded, going from 0 to 1.  It is also symmetric about 0.5.   In practice, this scale is most interpretable.
 

3)

**Comment by isabellaghement** [comment]
To me, in reading this post, it was NOT clear whether one could define an Average Treatment Effect (ATE) on the logit scale as well as on the probability scale.  I would like to know if that is possible?  

It would be helpful to the reader if you could state upfront that you will define the ATE directly on the probability scale.  

For the ANOVA-type model,  one can easily see that logit(p_i) = beta0 for the control treatment and logit(p_i) = beta0 + beta1 for the intervention treatment, where i = 1 to n.  Re-expressing this means that p_i = exp(beta0)/[1 + exp(beta0)] for the control treatment and  p_i = exp(beta0 + beta1)/[1 + exp(beta0 + beta1)] for the intervention treatment.  So if we want to take the average across i = 1 to n of exp(beta0)/[1 + exp(beta0)] and the average across i = 1 to n of  exp(beta0 + beta1)/[1 + exp(beta0 + beta1)] and then compute the difference of these two averages, do we get the ATE?  This was not clear to me and I think it should be clear to the reader.  

Sometimes the details provided on how to compute things in R seem to obscure what is actually being computed.  

WHAT is computed should be stated clearly upfront. Only after that one should clarify HOW it is computed.

**Comment by isabellaghement** [comment]
The statement *"This approach, however, DOES NOT convert beta1_hat (in the ANOVA-type model) into a difference in probability. This is not an average treatment effect, and sadly, it’s completely uninterpretable."* does NOT make sense to me - probably because it wasn't clarified to the reader upfront what beta1 represents in the ANOVA-type model.  

beta1 (and its estimated value) ARE interpretable on the logit-scale: 

beta1 represents the difference in the logit-transformed probability that a subject's medical record indicated any STI testing at 6 weeks between the intervention group and the control group. 

Are you trying to say that beta1 does NOT represent a difference in probabilities in the ANOVA-type model?  

I guess you could say that provided you explain what does represent this difference in probabilities: 

 exp(beta0 + beta1)/[1 + exp(beta0 + beta1)] - exp(beta0)/[1 + exp(beta0)]. 

So I think the statement I flagged needs to be toned down for the reader and the appropriate explanations added to the post:  you are focusing on expressing the effect of treatment for each subject on the probability of interest (rather than logit-transformed probability); with this focus, you cannot rely on beta1 to quantify this effect - instead, you will rely on exp(beta0 + beta1)/[1 + exp(beta0 + beta1)] - exp(beta0)/[1 + exp(beta0)].  So this last quantity needs to be computed for each subject in the study and then averaged across subjects to get the ATE.

**Comment by isabellaghement** [comment]
In the ANOVA-type model, is your estimand the average of exp(beta0 + beta1)/[1 + exp(beta0 + beta1)] - exp(beta0)/[1 + exp(beta0)]  across ALL population subjects represented by the n subjects included in the study sample? Can you clarify that?  Then your estimate is simply the average of exp(beta0 + beta1)/[1 + exp(beta0 + beta1)] - exp(beta0)/[1 + exp(beta0)] across just the n subjects in the study sample?  

It would be helpful to clarify to the reader what the estimand is *i.e., quantity we are trying to estimate) and what the estimate is.

**Comment by isabellaghement** [comment]
Can you clarify how the inverse logit function is defined?  Something like:  inverse_logit(p) = exp(p)/[1 + exp(p)].

**Comment by isabellaghement** [comment]
Who is N? Is it the sample size or the population size?  Personally, I used n to denote sample size.  Can you clarify this?

**Comment by isabellaghement** [comment]
In the plot where you show the counterfactual probabilities derived from the ANOVA-type model, am I correct that counterfactual would only refer to the probability corresponding to the treatment which was NOT received by the participant in question? 

For example, if participant # 1 received the control treatment, you could derive their actual probability under the control treatment and their counterfactual probability under the intervention treatment.  Then, you could use one colour (say, blue) for the actual probability and another colour (say, red) for the counterfactual probability.  

It seems to me that the actual colour scheme does not allow the reader to see which participant received which treatment so they can understand which probabilities were actual and which ones were counterfactual.

**Reply by ASKurz** [reply]
No, these are purely model-based counterfactuals.

**Reply by isabellaghement** [reply]
I always thought that **counterfactual** means *"relating to or expressing what has not happened or is not the case"*.  So if the subject did receive a particular treatment, the counterfactual would refer to the treatment that was NOT received.  

I am not familiar with the causal literature other than through your blog posts - for others like myself, it is worth clarifying what you mean by **counterfactual** for the purpose of constructing this particular plot for the ANOVA-type model (and the other like it for the ANCOVA-type model).

**Reply by ASKurz** [reply]
I was hoping the notation of $\hat p_i^1$ and $\hat p_i^0$ would help clarify those are all model-based, not a mixture of model-based and actual. 

Going back to your earlier statement, 

> if participant # 1 received the control treatment, you could derive their actual probability under the control treatment

I'm not so sure that's the case. When it comes to the data, we don't observe a probability for participant #1. Rather, we observe a value of `0` or `1`. I suppose you could describe those observed values as either $p_i = 0$ or $p_i = 1$, but that seems unusual, to me.

**Reply by isabellaghement** [reply]
I see where you are coming from now, Solomon.  Your explanations here make it clear that **all probabilities you derived are indeed model-based**.   But some correspond to the treatment that was actually received and others correspond to the treatment that was not received.  Would it be possible to blend these two ideas in your description?  

I guess what I was trying to say is not that the probabilities were directly observed but rather that some of the model-based probabilities corresponded to the actually received treatment and others to the not-received treatment.  

Not sure if it makes sense to refer to the *model-based probabilities corresponding to the actually received treatment* as *actual probabilities* and to the other ones as *counterfactual probabilities*, but that is what I had in mind.  I do not mean *actual* as *observed* here, but as *actually received treatment*.

**Reply by ASKurz** [reply]
Okay. I'll chew on it.

**Comment by isabellaghement** [comment]
Does the avg_comparisons() function computes the average across just the n subjects included in the sample?  Can this be clarified in the post?

**Reply by ASKurz** [reply]
Yes, the `avg_comparisons()` function computes the average across just the n subjects included in the sample data. What other n would you be thinking of?

**Reply by isabellaghement** [reply]
First of all, I was confused as to whether you used *n* or *N* to denote the number of subjects in your sample.  When you define your ANOVA and ANCOVa-type models and use the index *i* for the i-th observation, can you clarify that *i* runs from 1 to *n*, where *n* is the sample size?   Then, later on in the blog post, where you use *N*, can you clarify what that *N* stands for?

Second of all, my confusion about what avg_comparisons() does stems from my confusion about what your estimand is and what the corresponding estimate is.  The estimand computes the expectation over all subjects in the population, whereas the estimate computes the average over all subjects in the sample.  With this distinction clarified in the post, it's worth mentioning to the reader that avg_comparisons() is used to compute the estimate (not the estimand).

**Reply by ASKurz** [reply]
Ah, yes, the `avg_comparisons()` function computes the model-based estimate of the estimand.

**Reply by isabellaghement** [reply]
Great!  Maybe this could be clarified for the reader with a brief note, so that it is clear to them what exactly avg_comparisons() does.

**Comment by isabellaghement** [comment]
When you move to the section on **ATE for the ANCOVA**, can you clarify early on that we now have to compute conditional ATE on the probability scale, where the conditioning is done with respect to specific values of the covariates we are adjusting the treatment effect for in the ANCOVA-type model? 

Then add that the specific values of the covariates can be chosen in various ways:  the mean value of the covariates observed in the sample (or the median or the mode), the actual values of the covariates observed in the sample, etc. 

However, once these specific values were identified, the idea of averaging the resulting conditional treatment effect across subjects to get an average conditional treatment effect still holds.  

I think having some big-picture clarification like this upfront would be really helpful to the reader; maybe even clarify that "conditional effect" and "adjusted effect" are one and the same thing.

**Comment by isabellaghement** [comment]
In defining those "ugly" effects indexed by TEMM, etc., why not give small illustrative examples?  For example, if our ANCOVA model includes age (continuous) and gender (categorical nominal) covariates, we can set age to its average value observed in the data and gender to its mode (i.e., its most frequent category represented in the data). This is so much easier for the reader to digest immediately after you introduce that abstract notation with C and D for continuous and discrete covariates.

**Comment by isabellaghement** [comment]
When talking about the CATE effects, is it possible to somehow categorize these effects according to their magnitude?  It strikes me that presenting the distribution of these effects is only half of the battle won.  

The other half is to identify subgroups of subjects who have a "large", "moderate" or "small" (counterfactual) treatment response based on their covariate values so that those subgroups of subjects could be targeted more specifically.  

Even something like *"the majority of subjects who had CATE effects above this particular threshold happened to have this specific combination of covariates"* may be helpful.

**Comment by isabellaghement** [comment]
I confess that I do NOT understand who $p_0$ and $p_1$ are given that the model itself was stated in terms of $p_i$?  Where did $p_0$ and $p_1$ come from and what do they stand for?
